# Import Pustaka

In [1]:
import numpy as np
import sympy as sy
import pandas as pd
from scipy import integrate 

# Problems

* Diketahui toko buah X menjual 10 produk dengan label nama A-J. Data penjualan serta pendapatan total toko X selalu dicatat setiap harinya dan berikut data penjualan 10 hari terakhir:
| Hari ke- | A | B | C | D | E | F | G | H | I | J | Pendapatan |
|:---------:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:------------:|
|1|14|19|2|44|9|1|3|40|17|22|1402500|
|2|7|26|26|24|54|2|2|4|22|18|2325000|
|3|37|54|1|43|46|23|37|36|18|17|3307000|
|4|19|19|45|41|43|21|38|51|56|37|4077000|
|5|47|28|50|10|44|54|3|47|2|21|3998000|
|6|12|41|31|25|44|13|16|9|47|3|2674500|
|7|20|19|53|56|34|6|60|37|35|49|4014000|
|8|60|44|1|16|9|24|55|55|30|10|2754500|
|9|41|55|50|28|59|54|40|9|45|34|5511500|
|10|14|11|39|2|26|17|51|19|38|39|3338500|

* Menggunakan konsep integral, tentukan luas negara Swiss jika diketahui koordinat batas-batasnya

# Sumber Dataset

Dataset koordinat batas Swiss dapat diakses pada link berikut: https://raw.githubusercontent.com/fahmimnalfrzki/Dataset/main/Swiss_boundaries.txt?token=AEZDEHXBTS7OMF32RTYPKZDA2FELA

## Harga Masing-masing produk A - J

#### Membuat Matrix

In [2]:
A = np.array([ [14,19,2,44,9,1,3,40,17,22, 1402500],[7,26,26,24,54,2,2,4,22,18,2325000],[37,54,1,43,46,23,37,36,18,17,3307000],[19,19,45,41,43,21,38,51,56,37,4077000],[47,28,50,10,44,54,3,47,2,21,3998000],[12,41,31,25,44,13,16,9,47,3,2674500],[20,19,53,56,34,6,60,37,35,49,4014000],[60,44,1,16,9,24,55,55,30,10,2754500],[41,55,50,28,59,54,40,9,45,34,5511500],[14,11,39,2,26,17,51,19,38,39,3338500]])
A = sy.Matrix(A)
A

Matrix([
[14, 19,  2, 44,  9,  1,  3, 40, 17, 22, 1402500],
[ 7, 26, 26, 24, 54,  2,  2,  4, 22, 18, 2325000],
[37, 54,  1, 43, 46, 23, 37, 36, 18, 17, 3307000],
[19, 19, 45, 41, 43, 21, 38, 51, 56, 37, 4077000],
[47, 28, 50, 10, 44, 54,  3, 47,  2, 21, 3998000],
[12, 41, 31, 25, 44, 13, 16,  9, 47,  3, 2674500],
[20, 19, 53, 56, 34,  6, 60, 37, 35, 49, 4014000],
[60, 44,  1, 16,  9, 24, 55, 55, 30, 10, 2754500],
[41, 55, 50, 28, 59, 54, 40,  9, 45, 34, 5511500],
[14, 11, 39,  2, 26, 17, 51, 19, 38, 39, 3338500]])

#### Melakukan Kombinasi Linear dengan Operasi Basis Elementer 

In [3]:
A.rref()

(Matrix([
 [1, 0, 0, 0, 0, 0, 0, 0, 0, 0,  5000],
 [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 11000],
 [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 12500],
 [0, 0, 0, 1, 0, 0, 0, 0, 0, 0,  3500],
 [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 17000],
 [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 26000],
 [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 10000],
 [0, 0, 0, 0, 0, 0, 0, 1, 0, 0,  2000],
 [0, 0, 0, 0, 0, 0, 0, 0, 1, 0,  7500],
 [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 24000]]),
 (0, 1, 2, 3, 4, 5, 6, 7, 8, 9))

Dari hasil kombinasi linear dapat diketahui harga masing-masing produk adalah:

* Harga produk **A** adalah **Rp. 5000**,
* Harga produk **B** adalah **Rp. 11000**,
* Harga produk **C** adalah **Rp. 12500**,
* Harga produk **D** adalah **Rp. 3500**,
* Harga produk **E** adalah **Rp. 17000**,
* Harga produk **F** adalah **Rp. 26000**,
* Harga produk **G** adalah **Rp. 10000**,
* Harga produk **H** adalah **Rp. 2000**,
* Harga produk **I** adalah **Rp. 7500**,
* Harga produk **J** adalah **Rp. 24000**

menguji harga dengan memasukan harga masing - masing produk kedalam persamaan pertama 

In [4]:
14*5000 + 19*11000 + 2*12500 + 44* 3500 + 9*17000 + 1*26000+ 3*10000 + 40*2000 + 17*7500 + 22*24000

1402500

## Menentukan Luas Negara Swiss

#### Melakukan Data Loading 

In [5]:
df = pd.read_csv('https://raw.githubusercontent.com/fahmimnalfrzki/Dataset/main/Swiss_boundaries.txt?token=AEZDEHXBTS7OMF32RTYPKZDA2FELA')

#### Menampilkan dataset

In [6]:
df

longitude  latitude
0      9.56672  47.54045
1      9.55980  47.50209
2      9.59660  47.46944
3      9.64965  47.45861
4      9.67035  47.39069
..         ...       ...
273    9.03333  47.68833
274    9.08264  47.68541
275    9.26110  47.66284
276    9.47832  47.57585
277    9.56672  47.54045

[278 rows x 2 columns]

#### Menampilkan info dataset

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 278 entries, 0 to 277
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   longitude  278 non-null    float64
 1   latitude   278 non-null    float64
dtypes: float64(2)
memory usage: 4.5 KB


Keterangan :

* sumbu **x** merupakan **latitude**
* sumbu **y** merupakan **longitude**

#### Mencari titik x tertinggi

In [8]:
y_max = df['latitude'].max()
y_max

47.80666000000008

titik x tertinggi negara swiss adalah **47.80666000000008 latitude**

#### Mencari titik x terendah

In [9]:
y_min = df['latitude'].min()
y_min

45.82944000000003

titik x terendah negara swiss adalah **45.82944000000003 latitude**

#### Mencari titik y tertinggi

In [10]:
x_max = df['longitude'].max()
x_max 

10.488210000000038

titik y tertinggi negara swiss adalah **10.488210000000038 longitude**

#### Mencari titik y terendah

In [11]:
x_min = df['longitude'].min()
x_min

5.967010000000073

titik y terendah negara swiss adalah **5.967010000000073 longitude**

#### Rentang Panjang negara swiss

45.82944000000003 - 47.80666000000008 latitude

#### Rentang Panjang dan Fungsi North Swiss

In [12]:
x = np.linspace(45.82944000000003, 47.80666000000008)
f_north = 10.488210000000038*x 

#### Integral North Swiss

In [13]:
luas_north = integrate.trapz(f_north, x)
luas_north

970.8892452154909

Integral north swiss adalah **970.8892452154909**

#### Rentang Panjang dan Fungsi South Swiss

In [14]:
x = np.linspace(45.82944000000003, 47.80666000000008)
f_south = 5.967010000000073*x

#### Integral South Swiss

In [15]:
luas_south = integrate.trapz(f_south,x)
luas_south

552.3636383227772

Integral south swiss adalah **552.3636383227772**

## Luas Negara Swiss

In [16]:
luas_swiss = luas_north - luas_south
luas_swiss

418.52560689271365

Luas negara swiss secara keseluruhan adalah **418.52560689271365**